File: common/src/main/java/dev/obscuria/tooltips/client/element/frame/BlankFrame.java
package dev.obscuria.tooltips.client.element.frame;

import com.mojang.serialization.Codec;
import net.minecraft.client.gui.GuiGraphics;

public record BlankFrame() implements TooltipFrame {

    public static final BlankFrame INSTANCE = new BlankFrame();
    public static final Codec<BlankFrame> CODEC = Codec.unit(INSTANCE);

    @Override
    public Codec<BlankFrame> codec() {
        return CODEC;
    }

    @Override
    public void render(GuiGraphics graphics, int x, int y, int width, int height) {}
}

File: common/src/main/kotlin/dev/obscuria/tooltips/client/element/frame/BlankFrame.kt
package dev.obscuria.tooltips.client.element.frame

import com.mojang.serialization.Codec
import net.minecraft.client.gui.GuiGraphics

object BlankFrame: ITooltipFrame {

    val CODEC: Codec<BlankFrame> = Codec.unit(BlankFrame)

    override fun codec(): Codec<out ITooltipFrame> = CODEC

    override fun render(graphics: GuiGraphics, x: Int, y: Int, width: Int, height: Int) {}
}